src/Middleware/HttpLogging/src/RequestBufferingStream.cs

```diff
@@ -27,6 +27,12 @@ public override async ValueTask<int> ReadAsync(Memory<byte> destination, Cancell
     {
         var res = await _innerStream.ReadAsync(destination, cancellationToken);
 
+        // Zero-byte reads can occur when using PipeReader, we don't want to accidentally complete the RequestBody logging in this case
```

Comment:
From context, I assume passing zero bytes to `WriteToBuffer` would "complete the RequestBody logging"?